app/channel.go

```diff
@@ -63,13 +63,15 @@ func (a *App) JoinDefaultChannels(teamId string, user *model.User, channelRole s
 			l4g.Warn("Failed to update ChannelMemberHistory table %v", result.Err)
 		}
 
-		if requestor == nil {
-			if err := a.postJoinTeamMessage(user, townSquare); err != nil {
-				l4g.Error(utils.T("api.channel.post_user_add_remove_message_and_forget.error"), err)
-			}
-		} else {
-			if err := a.PostAddToTeamMessage(requestor, user, townSquare, ""); err != nil {
-				l4g.Error(utils.T("api.channel.post_user_add_remove_message_and_forget.error"), err)
+		if *a.Config().ServiceSettings.ExperimentalEnableDefaultChannelLeaveJoinMessages == true {
```

Comment:
Minor but this can just be `if *a.Config().ServiceSettings.ExperimentalEnableDefaultChannelLeaveJoinMessages {`